Applications/Spire/Source/SecurityInput/SecurityInputDialog.cpp

```diff
@@ -115,11 +118,24 @@ void SecurityInputDialog::mouseReleaseEvent(QMouseEvent* event) {
 }
 
 void SecurityInputDialog::showEvent(QShowEvent* event) {
-  m_security_input_box = new SecurityInputBox(Ref(*m_model), m_initial_text,
-    this);
-  m_security_input_box->connect_commit_signal(
-    [=] (const auto& s) { set_security(s); });
-  m_layout->addWidget(m_security_input_box);
+  m_security_line_edit = new SecurityInputLineEdit(m_initial_text,
+    Ref(*m_model), true, this);
+  m_security_line_edit->setFixedSize(scale(180, 30));
+  connect(m_security_line_edit, &SecurityInputLineEdit::editingFinished,
+    [=] { set_security(m_security_line_edit->get_security()); });
+  m_security_line_edit->installEventFilter(this);
+  m_security_list = m_security_line_edit->findChild<DropDownList*>();
+  m_security_list->installEventFilter(this);
+  m_layout->addWidget(m_security_line_edit);
+  m_security_line_edit->show();
+  m_security_line_edit->setFocus();
+}
+
+void SecurityInputDialog::reject_dialog() {
+  disconnect(m_security_line_edit,
+    &SecurityInputLineEdit::editingFinished, nullptr, nullptr);
+  reject();
+  static_cast<QWidget*>(parent())->activateWindow();
```

Comment:
What guarantee is there that a parent exists?

Reply:
There's no hard guarantee, but given our coding standards it should always have one. I can remove the default parameter to ensure it always a parent, but I'm not sure why this would be a special case.

Reply:
Perhaps a better question is why does this dialog call ```activateWindow``` on the parent? What would happen otherwise?

Reply:
I added this when after closing the SecurityInputDialog the tester window would receive focus instead of the parent Time and Sales window.

Reply:
Best to look into that issue directly. My understanding is that when a window closes, its parent is what gets activated by default.

Reply:
I can't reproduce it now, either on accepting or rejecting the SecurityInputDialog. It may have been a transient issue when updating the windows/widgets.

Reply:
What I'm seeing is that Qt processes the system event and receives information on which window to activate next based on that event. I'll look into this more tomorrow.